internal/mode/static/nginx/config/validation/http_filters.go

```diff
@@ -56,20 +49,12 @@ func (HTTPRedirectValidator) ValidateHostname(hostname string) error {
 
 // ValidateRewritePath validates a path used in a URL Rewrite filter.
 func (HTTPURLRewriteValidator) ValidateRewritePath(path string) error {
-	if path == "" {
-		return nil
-	}
-
-	if !pathRegexp.MatchString(path) {
-		msg := k8svalidation.RegexError(pathErrMsg, pathFmt, pathExamples...)
-		return errors.New(msg)
-	}
-
-	if strings.Contains(path, "$") {
-		return errors.New("cannot contain $")
-	}
+	return validatePath(path)
+}
 
-	return nil
+// ValidateRedirectPath validates a path used in a Request Redirect filter.
+func (HTTPRedirectValidator) ValidateRedirectPath(path string) error {
```

Comment:
Can this and ValidateRewritePath just be collapsed into a single function, `ValidatePath`?